pkg/clustersync/clusterSync.go

```diff
@@ -314,4 +324,47 @@ func processClusterDelete(ctx context.Context, obj interface{}) {
 		return
 	}
 	dao.DeleteClusterAndResources(ctx, clusterName, deleteClusterNode)
+
+}
+
+// finds lingering data in database from deleted/detached clusters or clusters with search-collector-addon disabled:
+func confirmDelete(ctx context.Context, dynamicClient dynamic.Interface, gvr schema.GroupVersionResource) ([]string, error) {
+	var needToDelete []string
+	managedClustersFromClient := make(map[string]struct{})
+
+	// get all managed clusters via dynamic client:
+	resourceObj, err := dynamicClient.Resource(gvr).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.Warning("Error resolving ManagedClusters with dynamic client", err.Error())
+		return nil, err
+	}
+	for _, item := range resourceObj.Items {
+		// Here we want all managed clusters that have the search-collector addon available
+		if item.GetName() != "local-cluster" && // TO DO: need better method instead of using name.
+			item.GetLabels()["feature.open-cluster-management.io/addon-search-collector"] == "available" {
+			managedClustersFromClient[item.GetName()] = struct{}{}
+		}
+	}
+	klog.V(3).Infof("Managed Clusters reported from kube client: ", managedClustersFromClient)
+
+	// get all managed clusters from db:
+	managedClustersFromDB, err := dao.GetManagedCluster(ctx)
+	if err != nil {
+		klog.Errorf("Error getting managed clusters names from database. %s", err)
+		return nil, err
+	} else {
+		klog.V(3).Infof("Managed Clusters reported from database: ", managedClustersFromDB)
+	}
+	for _, dmCluster := range managedClustersFromDB {
+		if _, exist := managedClustersFromClient[dmCluster]; !exist {
+			// At this point the cluster exists in DB, but not in the list from client.
+			needToDelete = append(needToDelete, dmCluster)
+			klog.V(3).Infof("Found Managed Cluster data in database that should be deleted! Cluster found: %s", dmCluster)
+		} else {
+			klog.V(5).Infof("Managed Cluster data successfully deleted from database.")
+		}
```

Comment:
- We can make this log V(1) because it should happen very infrequently.
- No need to log the else here. It would generate too many logs and we can infer it when the first log isn't present.
```suggestion
			klog.V(1).Infof("Found Managed Cluster data in database that should be deleted! Cluster found: %s", dmCluster)
		}
```